src/kimmdy/runmanager.py

```diff
@@ -718,35 +732,64 @@ def _decide_recipe(
 
         # FIXME: Hotfix for #355 aggregate not working for big systems
         if "rfkmc" != self.kmc_algorithm:
-            self.recipe_collection.aggregate_reactions()
+            for collection in self.recipe_collections.values():
```

Comment:
I think _decide_recipe looks super messy at the moment. Do we have too many edge cases for this?